src/composables/useWorkflowPersistence.ts

```diff
@@ -114,12 +115,15 @@ export function useWorkflowPersistence() {
   const storedActiveIndex = JSON.parse(
     getStorageValue('Comfy.ActiveWorkflowIndex') || '-1'
   )
-  watch(restoreState, ({ paths, activeIndex }) => {
-    setStorageValue('Comfy.OpenWorkflowsPaths', JSON.stringify(paths))
-    setStorageValue('Comfy.ActiveWorkflowIndex', JSON.stringify(activeIndex))
-  })
+  if (workflowPersistenceEnabled.value) {
```

Comment:
The check of setting value should be in the watch callback. Otherwise it won't persist value if setting value is flipped from off to on.